server/routes/messages.ts

```diff
@@ -15,138 +15,113 @@ async function connect() {
-            }
-            sendToQueueAndRespond(queueName, data, res)
+                .send('The content must contain the message property')
         }
+    } else {
+        return res.status(400).send('Invalid operationId')
+    }
+    sendAndRespond(data, res, queueName, topic)
+}
+
+function handleMicroserviceRoute(
+    routeName: string,
+    queueName: string,
+    validOperationIds: string[]
+) {
+    router.post(`/${routeName}`, (req: Request, res: Response) => {
+        validateAndSend(req, res, validOperationIds, queueName)
     })
 }
 
-handleMicroserviceRoute(
-    'users-microservice',
-    'users_microservice',
-    ['requestAppUsers', 'notificationNewPlanPayment', 'responseUserReviews'],
-    ['notificationNewPlanPayment', 'responseUserReviews']
-)
-handleMicroserviceRoute(
-    'mailbox-microservice',
-    'mailbox_microservice',
-    ['responseAppUsers'],
-    ['responseAppUsers']
-)
-handleMicroserviceRoute(
-    'reviews-microservice',
-    'reviews_microservice',
-    ['requestUserReviews', 'requestCourseReviews', 'requestMaterialReviews'],
-    ['requestUserReviews', 'requestCourseReviews', 'requestMaterialReviews']
-)
-handleMicroserviceRoute(
-    'courses-microservice',
-    'courses_microservice',
-    [
-        'publishNewAccess',
-        'responseAppClassesAndMaterials',
-        'responseCourseReviews',
-        'notificationNewClass',
-        'notificationDeleteClass',
-        'notificationAssociateMaterial',
-        'notificationDisassociateMaterial',
-    ],
-    [
-        'publishNewAccess',
-        'responseAppClassesAndMaterials',
-        'responseCourseReviews',
-        'notificationNewClass',
-        'notificationDeleteClass',
-        'notificationAssociateMaterial',
-        'notificationDisassociateMaterial',
-    ]
-)
-handleMicroserviceRoute(
-    'payment-history-microservice',
-    'payment_history_microservice',
-    ['notificationNewPayment'],
-    ['notificationNewPayment']
-)
-handleMicroserviceRoute(
-    'payment-microservice',
-    'payment_microservice',
-    ['notificationUserPlanInfo'],
-    ['notificationUserPlanInfo']
-)
-handleMicroserviceRoute(
-    'access-control-microservice',
-    'access_control_microservice',
-    ['notificationNewCourseOrMaterialBought'],
-    ['notificationNewCourseOrMaterialBought']
-)
-handleMicroserviceRoute(
-    'learning-microservice',
-    'learning_microservice',
-    [
-        'responseMaterialReviews',
-        'requestAppClassesAndMaterials',
-        'publishNewAccess',
-    ],
-    [
-        'responseMaterialReviews',
-        'requestAppClassesAndMaterials',
-        'publishNewAccess',
-    ]
-)
+router.post('/user/notification', (req: Request, res: Response) => {
+    validateAndSend(
+        req,
+        res,
+        ['notificationUserDeletion'],
+        undefined,
+        'userRemoved'
+    )
+})
+
+handleMicroserviceRoute('users-microservice', 'users_microservice', [
```

Comment:
Hasn't it been discussed in meetings that which users will need to consider a request to send the information of all users, or is it on another line and I haven't seen it?

Reply:
Absolutely right, @TomasCB24, it's already solved!